### PR TITLE
[EMCAL-647] Event builder for cell container from different subtimefr…

### PR DESCRIPTION
**Pull request by mfasDa**
…ames

Combining cells from multiple subevents in a single
cell / trigger record container according to cells
belonging to the same collsion. Output is sent to
the same output type as input cell / trigger record
container but with subspecification 0 for
compatibility with existing containers.

### CODE REVIEW

**Comment by mfasDa** [comment]
From #6834
> @mfasDa I also don't see reopen button, perhaps @ktf can do this. As for the merging, was it tested? I remember there was a problem of defining FLP-specific subspecs.

Not yet, as we are missing a test setup and for the pilot we had other priorities. 

For what concerns the FLP-specific subspec I have an implementation here: https://github.com/mfasDa/AliceO2/commit/c0b1101a75f0d44413b30653d1d98206032f4196 (#6772 - also stale-closed - reopened in #7642). It is not yet in a PR, it still needs to be tested on a dedicated test FLP.

File: Detectors/EMCAL/workflow/src/EventBuilderSpec.cxx [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+
+  auto posCells = ctx.inputs().getPos("inputcells"),
+       posTriggerRecords = ctx.inputs().getPos("inputtriggers");
+  auto numSlotsCells = ctx.inputs().getNofParts(posCells),
+       numSlotsTriggerRecords = ctx.inputs().getNofParts(posTriggerRecords);
+  for (decltype(numSlotsCells) islot = 0; islot < numSlotsCells; islot++) {
+    auto celldata = ctx.inputs().getByPos(posCells, islot);
+    auto subspecification = framework::DataRefUtils::getHeader<header::DataHeader*>(celldata)->subSpecification;
+    // Discard message if it is a deadbeaf message (empty timeframe)
+    if (subspecification == 0xDEADBEEF) {
```
Comment (shahor02) [inline Detectors/EMCAL/workflow/src/EventBuilderSpec.cxx:46]: see comment https://github.com/AliceO2Group/AliceO2/pull/7642/files#r752583346 
Why not sending with the standard subspec and skip if the payload is absent (which will happen if the raw data had 0xdeadbeef)?

**Review by shahor02 — COMMENTED** [review]
With so many map operations you may have overheads (I think T0 had these in an almost similar situation). It could be more efficient to 1st to record in the vector<int> the trigger_bc.toLong() - TF_first_IR.toLong() of every trigger, associating with every filled slot vectors of corresponding cells and in triggers, then process them.

**Comment by ktf** [comment]
This will be closed soon. @mfasDa is it still being worked on?

**Comment by shahor02** [comment]
Hi @ktf ,
This PR is waiting for the feature we were discussing: possibility to define wildcarded subspecs which does not match to one specific subspec, to avoid short circuits

**Comment by mfasDa** [comment]
Hi @ktf ,

Yes, this is needed, for a triggered detector we have to merge triggered events within a timeframe from the two subtimeframes manually once we move the reconstruction from the EPN to the FLP. However as @shahor02 explained there are some questions in the handling of input and output channels, i.e. related to the subspecification, and waiting for some features, and therefore the PR is on hold. It would be good if the pull request could be kept open, it was already closed once and reopened manually. In AliRoot we had the feature to make the PR a draft if it was not yet ready for merging, but I don't see the feature in O2, would be nice to add it here as well. The same applies to PR #7642 which is the corresponding on the sender side (FLP) which has to be adapted and merged in sync with this PR on the EPN (receiver side).

**Pull request closed without merging**